renderer/components/UI/CurrencyFieldGroup.js

```diff
@@ -30,30 +30,66 @@ const CurrencyFieldGroup = React.forwardRef(
       validate,
       validateOnBlur,
       validateOnChange,
+      onChange,
       ...rest
     },
     ref
   ) => {
+    // Whether update of a linked input (crypto<->fiat) is enabled.
+    // This ref is used in a process of linked value calculations to prevent circular updates.
+    const blockUpdates = useRef(false)
+
+    // Prevent updates of linked form values.
+    const blockLinkedUpdates = () => {
+      blockUpdates.current = true
+    }
+
+    // Reenable updates of linked form elements.
+    const unblockLinkedUpdates = async () => {
+      await Promise.resolve()
+      blockUpdates.current = false
+      onChange && onChange()
+    }
+
+    // Get current block status of linked form elements.
+    const isLinkedUpdatesBlocked = () => {
+      return blockUpdates.current
+    }
+
     /**
      * handleAmountCryptoChange - Set the amountFiat field whenever the crypto amount changes.
      *
-     * @param {Event} e Event
+     * @param {string} value Value
      */
-    const handleAmountCryptoChange = e => {
-      const lastPrice = currentTicker[fiatCurrency]
-      const value = convert(cryptoUnit, 'fiat', e.target.value, lastPrice)
-      formApi.setValue('amountFiat', value)
+    const handleAmountCryptoChange = async value => {
+      if (!isLinkedUpdatesBlocked()) {
+        const lastPrice = currentTicker[fiatCurrency]
+        const fiatValue = convert(cryptoUnit, 'fiat', value, lastPrice)
+        blockLinkedUpdates()
```

Comment:
Could I suggest moving this line up to the top of this block so that it reads more like:

```
if (!isLinkedUpdatesBlocked()) {
  blockLinkedUpdates()
  // do stuff ...
  await unblockLinkedUpdates()
}
```